### PR TITLE
Oauth overview

### PR DESCRIPTION
**Pull request by mystiberry-okta**
## Description:
- Oauth Overview's final destination not clear yet.

## Type of Pull Request:

- [x] Content (Documentation updates or typo-fixes)
- [ ] Blog Post
- [ ] Functional (CSS changes, JS updates, or layout modifications)

### Resolves:
* [OKTA-124731](https://oktainc.atlassian.net/browse/OKTA-124731)

## How Has This Been Tested?
- [x] I have built this locally and verified that it does not break existing functionality.
- [ ] For functional changes, I have tested against Chrome, Firefox, and Safari, and mobile.
- [ ] For functional changes, I have added tests.

## Primary Reviewers:
Fed Team--but we're not sure where to put this yet.

### CODE REVIEW

**Comment by mystiberry-okta** [comment]
working on link fixes.

File: _source/_docs/api/resources/oauth2-intro.md [hunk 1]
```diff
@@ -0,0 +1,320 @@
...
+For example, assume the following conditions are in effect:
+
+- Rule A has priority 1 and one people condition to include user U;
+- Rule B has priority 2 and one people condition to include all the users assigned to the client;
+
+Because Rule A has a higher priority, the requests for user U are evaluated in Rule A, and Rule B is not evaluated.
+
+The requests with `client_credentials` grant type match "no user" condition, which excludes everyone, because there is no user session for the request.
+
+The actions in a Rule define which scopes can be granted to the requests, thus determines which claims will be added into the tokens. They also define the lifetime of the Access Token and Refresh Token.
```
Comment (yuliu-okta) [inline _source/_docs/api/resources/oauth2-intro.md]: We have moved the scope from action to condition, so the action is just the lifetime.

File: _source/_standards/OAuth/index.md [hunk 2]
```diff
@@ -14,9 +14,324 @@ The specification and associated RFCs are developed by [the IETF OAuth WG](https
...
+Okta uses public key cryptography to sign tokens and verify that they are valid.
+
+The resource server must validate the Access Token before allowing the client to access protected resources.
+
+Access Tokens are sensitive and can be misused if intercepted. Transmit them only over HTTPS and only via POST data or within request headers. If you store them on your application, you must store them securely.
+
+An Access Token must be validated in the following manner:
+
+1. Verify that the `iss` (issuer) claim matches the identifier of your authorization server.
+2. Verify that the `aud` (audience) claim is the requested URL.
```
Comment (yuliu-okta) [inline _source/_standards/OAuth/index.md]: Verify that the `aud` (audience) claim is the value configured in AS

File: _source/_standards/OAuth/index.md [hunk 3]
```diff
@@ -14,9 +14,324 @@ The specification and associated RFCs are developed by [the IETF OAuth WG](https
...
+However, in an emergency situation you can still stay in sync with Okta's key rotation. Have your application check the `kid`, and if it has changed and the key is missing from the local cache, check the `jwks_uri` value in the [authorization server metadata](#authorization-server-metadata) and you can go back to the [jwks uri](/docs/api/resources/oauth2.html#get-keys) to get keys again from Okta
+
+Please note the following:
+
+* For security purposes, Okta automatically rotates keys used to sign the token.
+* The current key rotation schedule is four times a year. This schedule can change without notice.
+* In case of an emergency, Okta can rotate keys as needed.
+* Okta always publishes keys to the JWKS.
+* To save the network round trip, your app can cache the JWKS response locally. The standard HTTP caching headers are used and should be respected.
+{% beta %}
```
Comment (yuliu-okta) [inline _source/_standards/OAuth/index.md]: it can be removed

File: _source/_standards/OAuth/index.md [hunk 4]
```diff
@@ -14,9 +14,324 @@ The specification and associated RFCs are developed by [the IETF OAuth WG](https
...
+
+The lifetime of a Refresh Token is configured in [Access Policies](#access-policies), the minimum value is 24 hours. The refresh token can also expire after a period if no clients redeem it for an Access Token. The period should be equal to or larger than 10 minutes. If the token's lifetime is set to unlimited, the Authorization Server will not check if clients use it or not.
+
+### Refresh Token Revocation
+
+Refresh Tokens can be revoked explicitly by making a [Revocation Request](#revocation-request). Additionally, all Refresh Tokens associated with an entity are revoked when the entity is deactivated, deleted, or otherwise modified in a way that invalidates the associated Refresh Tokens. Such ways includes:
+
+* The User is Suspended or Deactivated
+* The Client App is Deactivated or Deleted
+* The Authorization Server's Resource URI is modified, or the Authorization Server is deleted.
```
Comment (yuliu-okta) [inline _source/_standards/OAuth/index.md]: audiences are modified

File: _source/_standards/OAuth/index.md [hunk 5]
```diff
@@ -14,9 +14,324 @@ The specification and associated RFCs are developed by [the IETF OAuth WG](https
...
+For example, assume the following conditions are in effect.
+
+- Policy A has priority 1 and applies to client C;
+- Policy B has priority 2 and applies to all the clients;
+
+Because Policy A has a higher priority, the requests coming from client C are evaluted by Policy A first. Policy B will not be evaluated if one of Rules in policy A matches the requests.
+
+### Rules
+
+In a policy the administrators can define several rules with people conditions. The people condition identifies users and groups that are included or exclueded to match the user the token is requested for. Rules are ordered numerically by priority. This priority determines the order in which they are searched for a user/group match. The highest priority rule has a priorityOrder of 1.
```
Comment (yuliu-okta) [inline _source/_standards/OAuth/index.md]: with people, scope and grant type conditions.

File: _source/_standards/OAuth/index.md [hunk 6]
```diff
@@ -14,9 +14,324 @@ The specification and associated RFCs are developed by [the IETF OAuth WG](https
...
+
+For example, assume the following conditions are in effect:
+
+- Rule A has priority 1 and one people condition to include user U;
+- Rule B has priority 2 and one people condition to include all the users assigned to the client;
+
+Because Rule A has a higher priority, the requests for user U are evaluated in Rule A, and Rule B is not evaluated.
+
+The requests with `client_credentials` grant type match "no user" condition, which excludes everyone, because there is no user session for the request.
```
Comment (yuliu-okta) [inline _source/_standards/OAuth/index.md]: we do not have "no user" condition now.

Comment (mystiberry-okta) [inline _source/_standards/OAuth/index.md]: I removed the whole sentence.

File: _source/_standards/OAuth/index.md [hunk 7]
```diff
@@ -14,9 +14,320 @@ The specification and associated RFCs are developed by [the IETF OAuth WG](https
...
+For example, assume the following conditions are in effect.
+
+- Policy A has priority 1 and applies to client C;
+- Policy B has priority 2 and applies to all the clients;
+
+Because Policy A has a higher priority, the requests coming from client C are evaluted by Policy A first. Policy B will not be evaluated if one of Rules in policy A matches the requests.
+
+### Rules
+
+In a policy the administrators can define several rules with people, scope, and grant type conditions. The people condition identifies users and groups that are included or exclueded to match the user the token is requested for. Rules are ordered numerically by priority. This priority determines the order in which they are searched for a user/group match. The highest priority rule has a priorityOrder of 1.
```
Comment (yuliu-okta) [inline _source/_standards/OAuth/index.md]: It's better that we can also add description for scope and grant type conditions.

Comment (mystiberry-okta) [inline _source/_standards/OAuth/index.md]: Of course--I'll add that today. Will create a JIRA with all the immediate follow on tasks here and in oauth2.

**Comment by cwu-okta** [comment]
We should link this to the OAuth CRUD endpoints page.

**Pull request merged**